uaclient/cli/fix.py

```diff
@@ -52,6 +104,747 @@ def fix_parser(parser):
     return parser
 
 
+def print_cve_header(cve: FixPlanResult):
+    lines = [
+        "{issue}: {description}".format(
+            issue=cve.title.upper(), description=cve.description
+        ),
+        " - https://ubuntu.com/security/{}".format(cve.title.upper()),
+    ]
+
+    print("\n".join(lines))
+
+
+def print_usn_header(fix_plan: FixPlanUSNResult):
+    target_usn = fix_plan.target_usn_plan
+    lines = [
+        "{issue}: {description}".format(
+            issue=target_usn.title.upper(), description=target_usn.description
+        ),
+    ]
+
+    additional_data = target_usn.additional_data
+    if isinstance(additional_data, USNAdditionalData):
+        if additional_data.associated_cves:
+            lines.append("Found CVEs:")
+            for cve in additional_data.associated_cves:
+                lines.append(" - https://ubuntu.com/security/{}".format(cve))
+        elif additional_data.associated_launchpad_bugs:
+            lines.append("Found Launchpad bugs:")
+            for lp_bug in additional_data.associated_launchpad_bugs:
+                lines.append(" - " + lp_bug)
+
+    print("\n".join(lines))
+
+
+def fix_cve(security_issue: str, dry_run: bool, cfg: UAConfig):
+    fix_plan = cve_plan(
+        options=CVEFixPlanOptions(cves=[security_issue]), cfg=cfg
+    )
+
+    error = fix_plan.cves_data.cves[0].error
+    if error:
+        raise exceptions.UserFacingError(error.msg)
+    print_cve_header(fix_plan.cves_data.cves[0])
+    print()
+
+    status, _ = execute_fix_plan(fix_plan.cves_data.cves[0], dry_run, cfg)
+    return status
+
+
+def fix_usn(
+    security_issue: str, dry_run: bool, no_related: bool, cfg: UAConfig
+):
+    fix_plan = usn_plan(
+        options=USNFixPlanOptions(usns=[security_issue]), cfg=cfg
+    )
+    error = fix_plan.usns_data.usns[0].target_usn_plan.error
+    if error:
+        raise exceptions.UserFacingError(error.msg)
```

Comment:
Shouldn't `usn_plan` raise exceptions itself? Though I guess it's too late to change now.

I'm forgetting - what class of errors get returned instead of raised from usn_plan?

Reply:
The only reason why we don't raise the exception directly from the API is because the user can provide multiple USNs/CVEs to be evaluated at the same time. Therefore, we try to create plans for as many as we can.

For the errors, we only raise errors if the USN name is either invalid or don't exist